src/app/preferences/general-preferences/general-preferences.service.ts

```diff
@@ -0,0 +1,21 @@
+namespace Core {
+  export class GeneralPreferencesService {
+    constructor(private $window: ng.IWindowService) {
+      'ngInject';
+    }
+    private static DEFAULT_VERTICAL_NAV_STATE = "show";
```

Comment:
fields should be put before constructor (not mandate syntactically but we prefer it for style consistency).

Reply:
also add `readonly` to make sure it's constant.